executor/builder.go

```diff
@@ -1997,6 +2018,12 @@ func buildNoRangeTableReader(b *executorBuilder, v *plannercore.PhysicalTableRea
 // buildTableReader builds a table reader executor. It first build a no range table reader,
 // and then update it ranges from table scan plan.
 func (b *executorBuilder) buildTableReader(v *plannercore.PhysicalTableReader) *TableReaderExecutor {
+	if b.ctx.GetSessionVars().TxnCtx.IsPessimistic && b.ctx.GetSessionVars().IsReadConsistencyTxn() {
```

Comment:
How about extract a method `IsPessimisticReadConsistency` for `SessionVars`.
Since `IsReadConsistencyTxn` is only used with `IsPessimistic`, we can inline it and remove this method.

Reply:
done